src/test/java/core/basesyntax/service/strategy/PurchaseHandlerTest.java

```diff
@@ -0,0 +1,49 @@
+package core.basesyntax.service.strategy;
+
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.exceptions.ProductQuantityException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class PurchaseHandlerTest {
+    private static final String FRUIT = "apple";
+    private static final int QUANTITY = 20;
+    private static final int SUBTRACT_QUANTITY = 5;
+    private final Storage storage = new Storage();
+    private BalanceHandler handler;
+    private PurchaseHandler purchaseHandler;
+
+    @BeforeEach
+    void setUp() {
+        handler = new BalanceHandler(storage);
+        purchaseHandler = new PurchaseHandler(storage);
+    }
+
+    @Test
+    void put_changedInformationToStorage_Ok() {
+        assertNull(storage.getData().get(FRUIT),
+                "Fruit should not exist in storage before operation.");
+        handler.operate(FRUIT, QUANTITY);
```

Comment:
do not use other handlers to populate the storage with the data that you need for a test, use direct storage access instead and put it there